server/src/testFixtures/java/io/crate/testing/SQLExecutor.java

```diff
@@ -839,19 +836,37 @@ public SQLExecutor startShards(String... indices) {
     }
 
     public SQLExecutor closeTable(String tableName) throws IOException {
-        String indexName = RelationName.of(QualifiedName.of(tableName), Schemas.DOC_SCHEMA_NAME).indexNameOrAlias();
+        RelationName relationName = RelationName.of(QualifiedName.of(tableName), Schemas.DOC_SCHEMA_NAME);
         ClusterState prevState = clusterService.state();
         var metadata = prevState.metadata();
-        String[] concreteIndices = IndexNameExpressionResolver
-            .concreteIndexNames(metadata, IndicesOptions.LENIENT_EXPAND_OPEN, indexName);
-
+        List<IndexMetadata> indices = metadata.getIndices(
+            relationName,
+            List.of(),
+            false,
+            imd -> imd.getState() == State.OPEN ? imd : null);
         Metadata.Builder mdBuilder = Metadata.builder(clusterService.state().metadata());
+        RelationMetadata relation = metadata.getRelation(relationName);
+        if (relation instanceof RelationMetadata.Table table) {
+            mdBuilder.setTable(
+                table.name(),
+                table.columns(),
+                table.settings(),
+                table.routingColumn(),
+                table.columnPolicy(),
+                table.pkConstraintName(),
+                table.checkConstraints(),
+                table.primaryKeys(),
+                table.partitionedBy(),
+                State.CLOSE,
+                table.indexUUIDs()
+            );
+        }
         ClusterBlocks.Builder blocksBuilder = ClusterBlocks.builder()
             .blocks(clusterService.state().blocks());
 
-        for (String index: concreteIndices) {
-            mdBuilder.put(IndexMetadata.builder(metadata.index(index)).state(IndexMetadata.State.CLOSE));
-            blocksBuilder.addIndexBlock(index, INDEX_CLOSED_BLOCK);
+        for (IndexMetadata imd: indices) {
+            mdBuilder.put(IndexMetadata.builder(imd).state(IndexMetadata.State.CLOSE));
```

Comment:
Shouldn't this only be applied if the relation is not an instanceof RelationMetadata.Table?
Not sure about the `indexBlock` below, must be applied in both cases probably.

Reply:
At least for partitioned tables it must also be done on index/partition level.

Can add some checks if you want - but I also have in mind to revisit some of this to see if we can re-use more logic from the tasks to avoid some duplication here and make sure unit tests have the same logic as the production code/integration tests use.

Reply:
Same wrong assumption of mine that `setTable` will create the needed IndexMetadata.